memory/vatest.py.data/va_test.c

```diff
@@ -125,240 +122,251 @@ int mmap_chunks_higher(unsigned long no_of_chunks, unsigned long hugetlb_arg)
-	printf("Allocating 16M chunks > 128TB \n");
-	mmap_chunks_higher(64, MAP_HUGETLB);
+	printf("Allocating hugepage chunks < 128TB \n");
+	mmap_chunks_lower(chunks, MAP_HUGETLB);
+}
+
+void alloc_huge_above_hint(int chunks)
+{
+	printf("Allocating hugepage chunks > 128TB \n");
+	mmap_chunks_higher(chunks, MAP_HUGETLB);
 }
 	
-void alloc_16g()
+void alloc_16g(int chunks)
 {
-	/* You need to configure 2 16G hugepages for this test to run */
-
 	printf("Allocating 16G chunks < 128TB \n");
-	mmap_chunks_lower(1, (MAP_HUGETLB | (34 << 26))); 
+	mmap_chunks_lower(chunks/2, (MAP_HUGETLB | (34 << 26)));
 
 	printf("Allocating 16G chunks > 128TB \n");
-	mmap_chunks_higher(1, (MAP_HUGETLB | (34 << 26)));
+	mmap_chunks_higher(chunks/2, (MAP_HUGETLB | (34 << 26)));
 }
 
-void alloc_16g_below_hint()
+void alloc_16g_below_hint(int chunks)
 {
-	/* You need to configure 1 16G hugepages for this test to run */
 	printf("Allocating 16G chunks < 128TB \n");
-	mmap_chunks_lower(1, (MAP_HUGETLB | (34 << 26))); 
-	
+	mmap_chunks_lower(chunks, (MAP_HUGETLB | (34 << 26)));
 }
 
-void alloc_16g_above_hint()
+void alloc_16g_above_hint(int chunks)
 {
-	/* You need to configure 1 16G hugepages for this test to run*/
 	printf("Allocating 16G chunks > 128TB \n");
-	mmap_chunks_higher(1, (MAP_HUGETLB | (34 << 26)));
+	mmap_chunks_higher(chunks, (MAP_HUGETLB | (34 << 26)));
 }
 
 /* <128Tb contains 8192 16Gb chunks
    >128Tb - 512Tb i.e 384Tb contains 24576 16Gb chunks */
 
 void fixed_position()
 {
-	  /* In this function we mmap 8000 1Tb slice for 64k pages,
-	  1Tb slice free for proper alignment, 
-	  1Tb slice for 16M hugepages , 1Tb slice for 16G hugepage 
-	  in both address ranges*/ 
-		
+	/* In this function we mmap 8000 1Tb slice for 64k pages,
+	1Tb slice free for proper alignment,
+	1Tb slice for 16M hugepages , 1Tb slice for 16G hugepage
+	in both address ranges*/
+
 	/* 8192 16Gb chunks < 128Tb*/
 	printf("Allocating 64k pages < 128TB \n");
-	mmap_chunks_lower(8000, 0);		
+	mmap_chunks_lower(8000, 0);
 	printf("Allocating 16M chunks < 128TB \n");
-	mmap_chunks_lower(64, MAP_HUGETLB);		
+	mmap_chunks_lower(64, MAP_HUGETLB);
 	printf("Allocating 16G chunks < 128TB \n");
-	mmap_chunks_lower(1, (MAP_HUGETLB | (34 << 26))); 
-	
+	mmap_chunks_lower(1, (MAP_HUGETLB | (34 << 26)));
+
 	/* 24576 16Gb chunks in 128Tb - 512Tb*/
 	printf("Allocating 64k pages > 128TB \n");
-	mmap_chunks_higher(24384, 0);		
+	mmap_chunks_higher(24384, 0);
 	printf("Allocating 16M chunks > 128TB \n");
-	mmap_chunks_higher(64, MAP_HUGETLB);		
+	mmap_chunks_higher(64, MAP_HUGETLB);
 	printf("Allocating 16G chunks > 128TB \n");
 	mmap_chunks_higher(1, (MAP_HUGETLB | (34 << 26)));
 }
 
 void mixed_position_16M()
 {
-	/*
-	  In this function we do try mmap 16M page 
-	  in between 64k and 16G is mapped at last 
-	  in both address ranges*/
-		
+	/* In this function we do try mmap 16M page
+	in between 64k and 16G is mapped at last
```

Comment:
can this be like `in between default and 16G`

Reply:
That part is yet to be fixed, as multiple hugepage needs tweak and last three scenarios are untouched. Still working on it.